tests/plugins/image2/resizerwithreadonly.js

```diff
@@ -1,54 +1,75 @@
+			CKEDITOR.addCss( css );
+		},
+
+		// #719, #2874
+		'test resizer creation without readOnly': testResizerVisibility( { name: 'editor1' } ),
 
-				bot.dialog( 'image', function( dialog ) {
-					dialog.setValueOf( 'info', 'src', '_assets/foo.png' );
-					dialog.getButton( 'ok' ).click();
+		// #719, #2874
+		'test resizer creation with readOnly at initialization': testResizerVisibility( { name: 'editor2', config: { readOnly: true } } ),
 
-					assert.sameData(
-						'cke_image_resizer',
-						editor.editable().getElementsByTag( 'span' ).$[ 2 ].outerHTML.match( /(cke_image_resizer)/g )[ 0 ]
-					);
+		// #719, #2874
+		'test resizer creation with readOnly after initialization': testResizerVisibility( { name: 'editor3' }, true ),
+
+		// #719, #2816
+		'test hovering in read-only': function() {
+			bender.editorBot.create( { name: 'editor4', config: { readOnly: true } } , function( bot ) {
+				var editor = bot.editor;
+				editor.setData( '<img src="_assets/foo.png" alt="" />', function() {
+					resume( function() {
+						var wrapper = editor.editable().findOne( '.cke_widget_wrapper' ),
+							resizer = editor.editable().findOne( '.cke_widget_image .cke_image_resizer' );
+						wrapper.addClass( 'hover' );
+						assert.isTrue( Boolean( !resizer.isVisible() ) );
+					} );
 				} );
+				wait();
 			} );
 		},
 
-		// #719
-		'test readOnly set to true': function() {
-			bender.editorBot.create( {
-				name: 'editor2',
-				config: {
-					readOnly: true
-				}
-			}, function( bot ) {
-				var editor = bot.editor,
-					listener;
-
-				listener = editor.on( 'dataReady', function() {
-					listener.removeListener();
-
+		// #719, #2816
+		'test hovering without read-only': function() {
+			bender.editorBot.create( { name: 'editor5', config: { readOnly: false } } , function( bot ) {
+				var editor = bot.editor;
+				editor.setData( '<img src="_assets/foo.png" alt="" />', function() {
 					resume( function() {
-						assert.areSame( 2, editor.editable().getElementsByTag( 'span' ).$.length );
+						var wrapper = editor.editable().findOne( '.cke_widget_wrapper' ),
+							resizer = editor.editable().findOne( '.cke_widget_image .cke_image_resizer' );
+						wrapper.addClass( 'hover' );
+						assert.isTrue( Boolean( resizer.isVisible() ) );
 					} );
 				} );
-
-				editor.setData( '<img src="_assets/foo.png" alt="" />' );
 				wait();
 			} );
 		}
 	} );
 }() );
+
+function testResizerVisibility( profile, readOnly ) {
+	return function() {
+		bender.editorBot.create( profile, function( bot ) {
+			var editor = bot.editor;
+
+			editor.setData( '<img src="_assets/foo.png" alt="" />', function() {
+				resume( function() {
+					if ( readOnly ) {
+						editor.setReadOnly( readOnly );
+					}
+					var resizer = editor.editable().findOne( '.cke_widget_image .cke_image_resizer' );
+					assert.isTrue( Boolean( resizer ), 'Resizer should be created'  );
```

Comment:
Better fit to check if element exists is to use `assert.isNotNull`. Also since `resizer` is used only once maybe it could be a one-liner:

```js
assert.isNotNull( editor.editable().findOne( '.cke_widget_image .cke_image_resizer' ), 'Resizer should be created' );
```